static/js/careers-filter-and-sort.js

```diff
@@ -5,42 +5,168 @@
   const noResults = document.querySelector(".js-filter__no-results");
   const jobContainer = document.querySelector(".js-filter-jobs-container");
   const sortSelect = document.querySelector(".js-sort");
+  const locationSelect = document.querySelector(".js-filter--location");
 
   var numberOfJobsDisplayed = 0;
-  var filters = [];
   var filterBy = {};
 
+  // Read data-location property and parse locations into well-defined categories
+  function parseLocations() {
+    const europe = ["emea", "slovakia", "bratislava", "europe", "uk", "germany", "berlin", "london", "worldwide"];
+    const americas = ["americas", "southwest", "san francisco", "usa", "austin", "texas", "tx", "brazil", "seattle", "america", "worldwide"];
+    const asia = ["apac", "taiwan", "taipei", "beijing", "china", "worldwide"];
+    const middleEast = ["emea", "worldwide"];
+    const africa = ["emea", "worldwide"];
+    const oceania = ["apac", "worldwide"];
+
+    const jobsList = document.querySelector(".js-job-list").children;
+
+    for(var n=0; n<jobsList.length; n++)
+    {
+      const location = jobsList[n].getAttribute("data-location");
+      var locationsList = "";
+
+      for(var x=0; x<europe.length; x++)
+      {
+        if(location.toLowerCase().includes(europe[x]))
+        {
+          locationsList += "europe ";
+          break;
+        }
+      }
+      for(var x=0; x<americas.length; x++)
+      {
+        if(location.toLowerCase().includes(americas[x]))
+        {
+          locationsList += "americas ";
+          break;
+        }
+      }
+      for(var x=0; x<asia.length; x++)
+      {
+        if(location.toLowerCase().includes(asia[x]))
+        {
+          locationsList += "asia ";
+          break;
+        }
+      }
+      for(var x=0; x<middleEast.length; x++)
+      {
+        if(location.toLowerCase().includes(middleEast[x]))
+        {
+          locationsList += "middle-east ";
+          break;
+        }
+      }
+      for(var x=0; x<africa.length; x++)
+      {
+        if(location.toLowerCase().includes(africa[x]))
+        {
+          locationsList += "africa ";
+          break;
+        }
+      }
+      for(var x=0; x<oceania.length; x++)
+      {
+        if(location.toLowerCase().includes(oceania[x]))
+        {
+          locationsList += "oceania ";
+          break;
+        }
+      }
+
+      if(locationsList.length > 0)
+      {
+        locationsList = locationsList.slice(0, locationsList.length-1);
+      }
+      jobsList[n].setAttribute("location-filter", locationsList);
+    }
```

Comment:
It's really good, works well - I had a suggestion that would reduce some repetition, see what you think:
```suggestion
const regions = {
        europe: ["emea", "slovakia", "bratislava", "europe", "uk", "germany", "berlin", "london", "worldwide"],
        americas: [ "americas", "southwest", "san francisco", "usa", "austin", "texas", "tx", "brazil", "seattle", "america", "worldwide"],
        asia: ["apac", "taiwan", "taipei", "beijing", "china", "worldwide"],
        "middle-east": ["emea", "worldwide"],
        africa: ["emea", "worldwide"],
        oceania: ["apac", "worldwide"],
      };

      const jobsList = document.querySelector(".js-job-list").children;

      for (var n = 0; n < jobsList.length; n++) {
        const location = jobsList[n].getAttribute("data-location");
        var locationsList = "";

        for (var region in regions) {
          var regionalLocations = regions[region];

          for (var i = 0; i < regionalLocations.length; i++) {
            if (location.toLowerCase().includes(regionalLocations[i])) {
              locationsList += region + " ";
            }
          }
        }

        if (locationsList.length > 0) {
          locationsList = locationsList.slice(0, locationsList.length - 1);
        }
        jobsList[n].setAttribute("location-filter", locationsList);
      }
```